### PR TITLE
fix(download): fixed Download Shared Encrypted File Should Work

### PR DESCRIPTION
**Pull request by cnlangzi**
- gosdk: https://github.com/0chain/gosdk/pull/373
- blobber: https://github.com/0chain/blobber/pull/536
- system_test: https://github.com/0chain/system_test/pull/193
- zbox: https://github.com/0chain/zboxcli/pull/169

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/encryption_header  |
| gosdk  | fix/encryption_header  |
| zboxcli  | fix/encryption_header  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | fix/encryption_header  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/encryption_header  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | fix/encryption_header  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | fix/encryption_header  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/encryption_header  |
| gosdk  | fix/encryption_header  |
| zboxcli  | fix/encryption_header  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | fix/encryption_header  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by stewartie4** [comment]
@cnlangzi Is it possible to add a test to check this new error?
https://github.com/0chain/gosdk/pull/373/files#diff-58fd13504400641409e5eaff205730ab3b97d39e35d81e9a39cd981d2a05a930R33

File: tests/cli_tests/zboxcli_share_file_test.go [hunk 1]
```diff
@@ -428,12 +427,16 @@ func TestShareFile(t *testing.T) {
 		downloadParams := createParams(map[string]interface{}{
 			"localpath":  file,
 			"authticket": authTicket,
+			"lookuphash": GetReferenceLookup(allocationID, file),
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_share_file_test.go:430]: can we add a test to download without the lookup hash?

**Review by stewartie4 — DISMISSED** [review]
some small coverage needed after new functionality added

File: tests/cli_tests/zboxcli_download_test.go [hunk 2]
```diff
@@ -416,18 +426,19 @@ func TestDownload(t *testing.T) {
...
 
-		// Just register a wallet so that we can work further
-		_, err := registerWallet(t, configPath)
-		require.Nil(t, err)
-
 		// Download file using auth-ticket: should work
-		output, err := downloadFile(t, configPath, createParams(map[string]interface{}{
+		output, err := downloadFileForWallet(t, viewerWalletName, configPath, createParams(map[string]interface{}{
 			"authticket": authTicket,
+			"lookuphash": GetReferenceLookup(allocationID, remotepath+filepath.Base(filename)),
```
[the remark this answers is not in the document]
Comment (cnlangzi) [reply tests/cli_tests/zboxcli_download_test.go:432]: > @cnlangzi Is it possible to add a test to check this new error? https://github.com/0chain/gosdk/pull/373/files#diff-58fd13504400641409e5eaff205730ab3b97d39e35d81e9a39cd981d2a05a930R33 image

hmmm, it is possible. but I am not sure if it worth to check it in system_test.

**Comment by cnlangzi** [comment]
> @cnlangzi Is it possible to add a test to check this new error? https://github.com/0chain/gosdk/pull/373/files#diff-58fd13504400641409e5eaff205730ab3b97d39e35d81e9a39cd981d2a05a930R33 image

hmmm, it is possible. but I am not sure if it worth to check it in system_test.

File: tests/cli_tests/zboxcli_download_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (cnlangzi) [reply tests/cli_tests/zboxcli_download_test.go:432]: do you think it is better to remove it? if yes, let me update them now

[the remark this answers is not in the document]
Comment (stewartie4) [reply tests/cli_tests/zboxcli_download_test.go:447]: nope keep it, t's good coverage 
but we should add an additional test that doesn't have

**Comment by stewartie4** [comment]
@cnlangzi Yes we should test it - it's worth having a test incase the code is accidentally changed some day - can you add one?

**Comment by cnlangzi** [comment]
> @cnlangzi Yes we should test it - it's worth having a test incase the code is accidentally changed some day - can you add one?

@stewartie4  done

**Review by stewartie4 — APPROVED** [review]

**Review by Kishan-Dhakan — APPROVED** [review]

**Pull request merged**